canvasapi/upload.py

```diff
@@ -39,8 +47,11 @@ def request_upload_token(self, file):
             and the JSON response from the API.
         :rtype: tuple
         """
-        self.kwargs["name"] = os.path.basename(file.name)
-        self.kwargs["size"] = os.fstat(file.fileno()).st_size
+        if isinstance(file, Path):
+            self.kwargs.update({"name": file.name, "size": file.stat().st_size})
```

Comment:
From what I can see, I think this code is unreachable with our current public API. We open the file before we call `get_upload_token`, so it's never a `Path`.

Reply:
I believe that's correct, this change can likely be reverted.
IIRC I added this check while debugging something and didn't remove it after.